### ISSUE TITLE
AJAX Request URL of project sometimes ends with "www.construct.net/[myFiles.json]"

### ISSUE THREAD

**Issue by jaydonteh** [Scirra/Construct-3-bugs#4463]
## Problem description

This is frankly quite hard to reproduce on my end but recently my projects that are using the AJAX request URL to fetch local project json files are getting errors. This happens in preview, not sure about export.

The error proceeds to stay even on multiple requests of varying json files. It continues to fail until I restart the preview or bad luck, I got into this bug again and no json files are fetched.

Looking into f12 inspection, it just says [Construct 3] Failed to parse JSON. And I saw my json project files are in the URL: "www.construct.net/[myFiles.json]", which probably is the cause.

## Steps to reproduce

1. Create a simple array json file in project folder "Files".
2. Add AJAX object and AJAX request the project file on a mouse click.
3. Preview.

## Observed result

Ajax returned error occasionally.

## Expected result

Successfully fetch data consistently. 

## More details

**Affected browsers/platforms:** 
MacOS and Windows, both Chrome and on NWJS.

**First affected release:** 
A few releases ago.

## System details

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 87.0.4280.141
Browser engine: Chromium
Browser architecture: 64-bit
Context: webapp
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1.25
Logical CPU cores: 8
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36
C3 release: r230 (beta)
Language setting: en-US

Local storage
Storage quota (approx): 70 gb
Storage usage (approx): 333 mb (0.5%)
Persistant storage: Yes

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

UI effects are disabled in settings.
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: Google Inc.
Renderer: ANGLE (Intel(R) HD Graphics 4600 Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_compression_bptc
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
EXT_texture_norm16
KHR_parallel_shader_compile
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBGL_multi_draw
OVR_multiview2
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm; codecs=opus)
Ogg Opus (audio/ogg; codecs=opus)
WebM Vorbis (audio/webm; codecs=vorbis)
Ogg Vorbis (audio/ogg; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM AV1 (video/webm; codecs=av01.0.00M.08)
MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
Please always follow the guidelines and attach a project file.

**Comment by jaydonteh** [comment]
@AshleyScirra I recently got the bug again and I want to get to the bottom of this. Here is an attached file that replicates how I coded in my own project.

I've also added the log file saved from DevTools F12 in Chrome when the bug happened. Multiple occurrence on "_TriggerError" 404.
 
Both the project and the log is in the archive file.
[Archive.zip](https://github.com/Scirra/Construct-3-bugs/files/5860842/Archive.zip)

Again, it happens occasionally and it was not easy to get it to browser -> alert "fail" but it did. Hope this issue gets addressed. I apologise for not adding a file initially as I wasn't sure how to give a file that can produce the error consistently and my time was limited.
